internal/controller/http/correlation/middleware.go

```diff
@@ -27,25 +30,25 @@ func ManageHeader(next http.Handler) http.Handler {
 	})
 }
 
-func OnResponseComplete(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		begin := time.Now()
-		next.ServeHTTP(w, r)
-		correlationId := IdFromContext(r.Context())
-		lc := LoggingClientFromContext(r.Context())
-		if lc != nil {
+func LoggingMiddleware(lc logger.LoggingClient) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			begin := time.Now()
+			correlationId := IdFromContext(r.Context())
+			lc.Trace("Begin request", clients.CorrelationHeader, correlationId, "path", r.URL.Path)
+			next.ServeHTTP(w, r)
 			lc.Trace("Response complete", clients.CorrelationHeader, correlationId, "duration", time.Since(begin).String())
-		}
-	})
+		})
+	}
 }
 
-func OnRequestBegin(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		correlationId := IdFromContext(r.Context())
-		lc := LoggingClientFromContext(r.Context())
-		if lc != nil {
-			lc.Trace("Begin request", clients.CorrelationHeader, correlationId, "path", r.URL.Path)
-		}
-		next.ServeHTTP(w, r)
-	})
+func RequestLimitMiddleware(n int64) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if strings.HasPrefix(r.URL.Path, v2.ApiDeviceRoute) && n > 0 {
```

Comment:
Does this really need to only be applied to `ApiDeviceRoute`? Other routes don't need it, but doesn't hurt to have it??

Reply:
I think the concern is the callback route, if the user accidentally set this value too low the callback would fail.

Reply:
Per @iain-anderson's comment, `http.MaxBytesReader` needs to be applied to all routes.
https://github.com/edgexfoundry/device-sdk-go/pull/878/files#r614019910